Documentation/Evolution/StronglyTypedCaptures.md

```diff
@@ -562,102 +536,133 @@ type safety to make its capture type be a homogeneous tuple instead of an array,
 
 ```swift
-let identifier = /[_a-zA-Z]+[_a-zA-Z0-9]*/  // => `Regex<Void>`
-print(str.firstMatch(of: identifier)?.captures)
-// Prints `()`.
+let regex = /ab(cd*)(ef)gh/
+// => `Regex<(Substring, Substring)>`
 ```
 
-`()` is also just a more consistent, continuous representation of a type with 0
-captures:
+However, it came with a number of (not necessarily insurmountable) open
+questions:
 
-| Number | Capture type             |
-|--------|--------------------------|
-| 0      | `()`                     |
-| 1      | `(Substring)`            |
-| 2      | `(Substring, Substring)` |
+- Will variadic generic tuple splatting preserve element labels?
+- Will variadic generic tuple splatting eliminate `Void`s? _(We don't want
+`firstMatch(of:)` to return `(Substring, Void)` for a regex with no captures)._
+- Will we be able to add [single-element labeled tuples](https://forums.swift.org/t/single-element-labeled-tuples/9797)?
+_(This would be needed to preserve the name of a capture in a regex with a
+single named capturing group.)_
+- What should be the type of `Captures` for a regex with no captures (e.g.
+`Void` or `Never` or something else)?
 
-### `Regex<(EntireMatch, Capture...)>` instead of `Regex<Captures>`
+Also, when using a regex within the result builder syntax, it's more obvious
+that `capture()` will have access to the entire match, when the match is
+included in the generic signature.
 
-There are a few downsides when `Regex`'s generic parameter represents captures:
-- When there is only one capture and it is a named one, e.g.
-  `\d{4}-(?<month>\d{2})-\d{2}`, the name will be lost in the type
-  `Regex<Substring>` because Swift does not support single-element tuples.
-- The indices in the `Captures` tuple do not align with the conventional
-  backreference indices, where the latter uses `\0` to refer to the entire match
-  and `\1` for the start of captures.
-  
-One alternative design to address these downsides is to include the whole match
-in the generic parameter. When a regex has captures, the generic argument is a
-tuple that can be seen as `(EntireMatch, Capture...)`. Capturing can then be
-seen as appending substrings on to an existing `Match` (as additional tuple
-elements).
+Given all of this, it seems simpler and more pragmatic to make `Regex` generic
+over both the match and the captures.
+
+### Structured rather than flat captures
+
+This pitch proposes infering capture types in such a way as to align with the
+traditional numbering of backreferences. This is because much of the motivation
+behind providing regex literals in Swift is their familiarity.
+
+If we decided to deprioritize this motivation, there are opportunities to infer
+safer, more ergonomic, and arguably more intuitive types for captures.
+
+For example, to be consistent with traditional regex backreferences
+quantifications of multiple or nested captures produces parallel arrays rather
+than an array of tuples.
```

Comment:
```suggestion
For example, to be consistent with traditional regex backreferences
quantifications of multiple or nested captures had to produce parallel arrays
rather than an array of tuples.
```